core/Piranha.AspNetCore/Models/SinglePageWithComments.cs

```diff
@@ -87,10 +87,12 @@ public override async Task<IActionResult> OnGet(Guid id, bool draft = false)
         /// <param name="draft">If the draft should be fetched</param>
         public virtual async Task<IActionResult> OnPostSaveComment(Guid id, bool draft = false)
         {
+            var ipAddress = Request.HttpContext.Connection.RemoteIpAddress;
+
             // Create the comment
             var comment = new Comment
             {
-                IpAddress = Request.HttpContext.Connection.RemoteIpAddress.ToString(),
+                IpAddress = ipAddress == null ? string.Empty : ipAddress.ToString(),
```

Comment:
Not actually part of the feature, but it seems like a good idea to do null check here.